engine/include/cubos/engine/ui/canvas/draw_type.hpp

```diff
@@ -0,0 +1,28 @@
+/// @file
+/// @brief Component @ref cubos::engine::UICanvas.
+/// @ingroup ui-canvas-plugin
+
+#pragma once
+
+#include <glm/glm.hpp>
+
+#include <cubos/core/gl/render_device.hpp>
+
+#include <cubos/engine/api.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Describes a UI draw call type.
+    /// @ingroup ui-canvas-plugin
+    struct CUBOS_ENGINE_API UIDrawType
+    {
+        static constexpr int MaxTextures = 4;
+
+        cubos::core::gl::ShaderPipeline pipeline;       ///< Handle to shader pipeline.
+        cubos::core::gl::ConstantBuffer constantBuffer; ///< Handle to the constant buffer to hold the per element data.
+        cubos::core::gl::ShaderBindingPoint constantBufferBindingPoint; ///< Binding point of the element data.
+        size_t perElementSize; ///< Size of the struct holding the element data.
+        cubos::core::gl::ShaderBindingPoint texBindingPoint[MaxTextures]; ///< Textures binding points.
+        size_t numTextures = 0;                                           ///< Number of textures the shader requires.
```

Comment:
```suggestion
```
You can replace this by just checking if the respective binding point is null on the canvas rendering.
Simplifies a bit the things on the user's side, as they don't have to manage the `numTextures`, at the cost of having to iterate over all of the 4 bindings points in the canvas side. The performance penalty shouldn't be a problem though.